test/nullable.jl

```diff
@@ -388,6 +393,119 @@ end
+@test @inferred(map(+, Nullable(0.0)))       === Nullable(0.0)
+@test @inferred(map(-, Nullable(1.0)))       === Nullable(-1.0)
+@test @inferred(map(-, Nullable{Float64}())) |> isnull_oftype(Float64)
+@test @inferred(map(sin, Nullable(1)))       === Nullable(sin(1))
+@test @inferred(map(sin, Nullable{Int}()))   |> isnull_oftype(Float64)
+
+# should not throw if function wouldn't be called
+@test map(x -> x ? 0 : 0.0, Nullable())       |> isnull_oftype(Union{})
+@test map(x -> x ? 0 : 0.0, Nullable(true))   === Nullable(0)
+@test map(x -> x ? 0 : 0.0, Nullable(false))  === Nullable(0.0)
+@test map(x -> x ? 0 : 0.0, Nullable{Bool}()) |> isnull_oftype(Union{})
+
+# broadcast and elementwise
+@test sin.(Nullable(0.0))            === Nullable(0.0)
+@test sin.(Nullable{Float64}())      |> isnull_oftype(Float64)
+@test @inferred(broadcast(sin, Nullable(0.0)))       === Nullable(0.0)
+@test @inferred(broadcast(sin, Nullable{Float64}())) |> isnull_oftype(Float64)
+
+@test Nullable(8) .+ Nullable(10)     === Nullable(18)
+@test Nullable(8) .- Nullable(10)     === Nullable(-2)
+@test Nullable(8) .+ Nullable{Int}()  |> isnull_oftype(Int)
+@test Nullable{Int}() .- Nullable(10) |> isnull_oftype(Int)
+
+@test log.(10, Nullable(1.0))                 === Nullable(0.0)
+@test log.(10, Nullable{Float64}())           |> isnull_oftype(Float64)
+@test log.(Nullable(10), Nullable(1.0))       === Nullable(0.0)
+@test log.(Nullable(10), Nullable{Float64}()) |> isnull_oftype(Float64)
+
+@test Nullable(2) .^ Nullable(4)      === Nullable(16)
+@test Nullable(2) .^ Nullable{Int}()  |> isnull_oftype(Int)
+
+# multi-arg broadcast
+@test Nullable(1) .+ Nullable(1) .+ Nullable(1) .+ Nullable(1) .+ Nullable(1) .+
+    Nullable(1) === Nullable(6)
+@test Nullable(1) .+ Nullable(1) .+ Nullable(1) .+ Nullable{Int}() .+
+    Nullable(1) .+ Nullable(1) |> isnull_oftype(Int)
+
+# these are not inferrable because there are too many arguments
+us = map(Nullable, 1:20)
+@test broadcast(max, us...) === Nullable(20)
+@test isnull(broadcast(max, us..., Nullable{Int}()))
+
+# test all elementwise operations
+for (eop, op) in ((.+, +), (.-, -), (.*, *), (./, /), (.\, \), (.//, //),
+                  (.==, ==), (.<, <), (.!=, !=), (.<=, <=), (.÷, ÷), (.%, %),
+                  (.<<, <<), (.>>, >>), (.^, ^))
+    # op(1, 1) chosen because it works for all operations
+    res = op(1, 1)
+    @test eop(Nullable(1), Nullable(1))         === Nullable(res)
+    @test eop(Nullable{Int}(), Nullable(1))     |> isnull_oftype(typeof(res))
+    @test eop(Nullable(1), Nullable{Int}())     |> isnull_oftype(typeof(res))
+    @test eop(Nullable{Int}(), Nullable{Int}()) |> isnull_oftype(typeof(res))
+    @test eop(Nullable(1), 1)                   === Nullable(res)
+    @test eop(1, Nullable(1))                   === Nullable(res)
+end
+
+# test reasonable results for Union{}
+# the exact types of these is finnicky and depends on implementation details
```

Comment:
Running inference on `Union{}` returns really unpredictable results, and I think would be something that changes frequently.